org-code-javabuilder/lib/src/main/java/org/code/javabuilder/LambdaRequestHandler.java

```diff
@@ -73,48 +79,54 @@ public String handleRequest(Map<String, String> lambdaInput, Context context) {
 
     final AmazonS3 s3Client = AmazonS3ClientBuilder.standard().build();
     final AWSFileWriter fileWriter =
-        new AWSFileWriter(s3Client, outputBucketName, javabuilderSessionId, getOutputUrl, logger);
+        new AWSFileWriter(s3Client, outputBucketName, javabuilderSessionId, getOutputUrl);
+    GlobalProtocol.create(
+        outputAdapter, inputAdapter, dashboardHostname, channelId, levelId, fileWriter);
+
+    // Create file loader
+    final UserProjectFileLoader userProjectFileLoader =
+        new UserProjectFileLoader(
+            GlobalProtocol.getInstance().generateSourcesUrl(),
+            levelId,
+            dashboardHostname,
+            useNeighborhood);
 
-    // Load files to memory and create and invoke the code execution environment
     try {
-      GlobalProtocol.create(
-          outputAdapter, inputAdapter, dashboardHostname, channelId, levelId, fileWriter);
+      // Delete any leftover contents of the tmp folder from previous lambda invocations
+      Util.recursivelyClearDirectory(Paths.get(System.getProperty("java.io.tmpdir")));
+    } catch (IOException e) {
+      // Wrap this in our error type so we can log it and tell the user.
+      InternalServerError error = new InternalServerError(INTERNAL_EXCEPTION, e);
 
-      try {
-        // Delete any leftover contents of the tmp folder from previous lambda invocations
-        Util.recursivelyClearDirectory(Paths.get(System.getProperty("java.io.tmpdir")));
-      } catch (IOException e) {
-        throw new InternalServerError(INTERNAL_EXCEPTION, e);
-      }
+      // Log the error
+      JSONObject eventData = new JSONObject();
+      eventData.put("exceptionMessage", error.getExceptionMessage());
+      eventData.put("loggingString", error.getLoggingString());
+      Logger.getLogger(MAIN_LOGGER).severe(eventData.toString());
```

Comment:
nit: seeing this in a couple of places, might be worth breaking it out into a method at some point (maybe adding a `String generateEventData()` to `JavabuilderThrowableProtocol`)? don't think it's required for this change though.

Reply:
That's a good idea. I'm going to merge as-is and then I may do a follow-up PR.